src/Runner.Sdk/Util/WhichUtil.cs

```diff
@@ -128,5 +128,13 @@ public static string Which(string command, bool require = false, ITraceWriter tr
 
             return null;
         }
+
+        // checks if the file is a symlink and if the symlink`s target exists.
+        private static bool IsSymlinkValid(string path)
```

Comment:
To me, this might be misleading, since if the file not symlink is passed, the name suggests that the method is operating on the symlink.
I would rather have method being `AssertFileExists` for example. The method would check if file is symlink.
If it is a symlink, checks if the original file exists and throws.
If the file is not a symlink and exists, just returns.
What do you think @fhammerl ?

Reply:
Yes, I think you are right about it being misleading. just to clarify, as I understood from the issue #2150, in case the symlink`s target is broken we should just ignore it, not throw an error.

> **High level goal**
Make the runner work the same as which. which ignores broken symlinks, while the runner treats them as if they were executables.

Reply:
changed to `IsPathValid`